pkg/distributor/distributorerror/error.go

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import "github.com/grafana/mimir/pkg/util/validation"
+
+type Error interface {
```

Comment:
I'm not a fan of another `Error` interface with a method that does nothing. I see it's used to distinguish between "distributor" and "ingesters" errors, but still. Interface should define a behaviour.

Reply:
After reviewing this PR 3 times, I agree here. There's no need for this interface. Distributor returns `error`, and some of them may have special meaning like `distributorerror.TooManyClusters`.

Reply:
I have removed the `distributorerror.Error` interface.